src/probnum/problems/_problems.py

```diff
@@ -24,27 +24,80 @@ class RegressionProblem:
         Observations of the latent process.
     locations
         Grid-points on which the observations were taken.
+    measurement_models
+        Measurement models.
     solution
         Array containing solution to the problem at ``locations``. Used for testing and benchmarking.
 
     Examples
     --------
+    >>> import numpy as np
+    >>> from probnum import statespace
     >>> obs = [11.4123, -15.5123]
     >>> loc = [0.1, 0.2]
-    >>> rp = RegressionProblem(observations=obs, locations=loc)
+    >>> model = statespace.DiscreteLTIGaussian(np.ones((1, 1)), np.ones(1), np.ones((1,1)))
+    >>> measurement_models = [model, model]
+    >>> rp = TimeSeriesRegressionProblem(observations=obs, locations=loc, measurement_models=measurement_models)
     >>> rp
-    RegressionProblem(observations=[11.4123, -15.5123], locations=[0.1, 0.2], solution=None)
+    TimeSeriesRegressionProblem(locations=[0.1, 0.2], observations=[11.4123, -15.5123], measurement_models=[DiscreteLTIGaussian(input_dim=1, output_dim=1), DiscreteLTIGaussian(input_dim=1, output_dim=1)], solution=None)
     >>> rp.observations
     [11.4123, -15.5123]
+
+    Regression problems are also indexable.
+
+    >>> len(rp)
+    2
+    >>> rp[0]
+    (0.1, 11.4123, DiscreteLTIGaussian(input_dim=1, output_dim=1))
     """
 
-    observations: np.ndarray
-    locations: np.ndarray
+    # The types are 'Sequence' (e.g. lists, tuples) or 'ndarray',
+    # because we need __len__ and __getitem__, which both provide.
+    # 'ndarray's are not 'Sequence's: https://github.com/numpy/numpy/issues/2776
+    locations: Union[Sequence, np.ndarray]
+    observations: Union[Sequence, np.ndarray]
+    measurement_models: Union[Sequence, np.ndarray]
 
-    # Optional: ground truth for testing and benchmarking
-    solution: typing.Optional[
-        typing.Callable[[np.ndarray], typing.Union[float, np.ndarray]]
-    ] = None
+    # For testing and benchmarking
+    # The solution here is a Sequence or array of the values of the truth at the location.
+    solution: Optional[Union[Sequence, np.ndarray]] = None
+
+    def __post_init__(self):
+        """Some postprocessing of the time-series regression problem inits.
+
+        1. Wrap the measurement models into an iterable of measurement models (by
+        default, a list).
+        2. Check that all inputs have the same length.
+        """
+
+        # If a single measurement model has been provided, transform it into a list of models
+        # to ensure that there is a measurement model for every (t, y) combination.
+        if not isinstance(self.measurement_models, abc.Iterable):
+            self.measurement_models = [self.measurement_models] * len(self.locations)
```

Comment:
Should we notify the user about this? Or are we fine with this side-effect just happening in the background? (asking just to be sure, I think it's fine, since we did it before, as well, just in the `filter` call

Reply:
I think for now we leave it. If eventually, we decide to deprecate the non-iterable argument, we can still do it then?